ext/js/display/display.js

```diff
@@ -1112,13 +1113,64 @@ class Display extends EventDispatcher {
                 }
                 button.disabled = !canAdd;
                 button.hidden = false;
+
+
+                const showTags = this._options.anki.showTags;
+                if (showTags && noteInfos) {
+                    this._setupTagsIndicator(i, noteInfos);
+                }
             }
             if (noteId !== null) {
                 this._viewerButtonShow(i, noteId);
             }
         }
     }
 
+    _setupTagsIndicator(i, noteInfos) {
+        const tagsIndicator = this._tagsIndicatorFind(i);
+        if (tagsIndicator === null) {
+            return;
+        }
+
+        let optionTags = this._options.anki.tags;
+        if (!Array.isArray(optionTags)) {
+            optionTags = [];
+        }
+
+        // in case a user has disabled note duplication check we look at the tags for all notes
+        // that were found. this seems to fit the purpose of showing tags better
+        // (i already have this word added but i still want to pay special attention to it for some reason)
+        let tags = noteInfos.reduce((ts, noteInfo) => ts.concat(noteInfo.tags), []);
```

Comment:
Since this is a highly-traversed code path, we should optimize the tags intersection check, something like this:
```js
const noteTags = new Set();
for (const {tags} of noteInfos) {
  for (const tag of tags) {
    noteTags.add(tag);
  }
}
for (const tag of optionTags) {
  noteTags.delete(tag);
}
```

(Rationale: standard for..of loops with a `Set` should be faster than `Array`'s `reduce/concat/filter/includes`, and use less memory)